packages/aws-cdk-lib/core/lib/nodejs-aspect.ts

```diff
@@ -0,0 +1,138 @@
+   */
+  public readonly targetRuntime: string;
+
+  constructor(runtime: string) {
+    this.targetRuntime = runtime;
+  }
+
+  public visit(construct: IConstruct): void {
+
+    //To handle custom providers
+    if (construct instanceof cdk.CustomResourceProviderBase) {
+      this.handleCustomResourceProvider(construct);
+    }
+
+    //To handle providers
+    else if (construct instanceof Provider) {
+      this.handleProvider(construct);
+    }
+
+    //To handle single Function case
+    else if (construct instanceof ReceiptRuleSet) {
+      this.handleReceiptRuleSet(construct);
+    }
+  }
+
+  private handleCustomResourceProvider(provider: cdk.CustomResourceProviderBase): void {
+    const node = provider as cdk.CustomResourceProviderBase;
+    // only replace for nodejs case
+    if (this.isValidRuntime(node.runtime)) {
+      const targetnode = node.node.children.find((child) => child instanceof cdk.CfnResource &&
+      child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+      targetnode.addPropertyOverride('Runtime', 'nodejs20.x');
+    }
+  }
+
+  private handleProvider(provider: Provider): void {
+    const functionNodes = provider.node.children.filter((child) => child instanceof lambda.Function);
+    for ( var node of functionNodes) {
+      const targetNode = node.node.children.find((child) => child instanceof cdk.CfnResource && child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        targetNode.addPropertyOverride('Runtime', 'nodejs20.x');
+      }
+
+      // onEvent Handlers
+      const onEventHandler = provider.onEventHandler as lambda.Function;
+      const onEventHandlerRuntime = onEventHandler.node.children.find((child) => child instanceof cdk.CfnResource && child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        onEventHandlerRuntime.addPropertyOverride('Runtime', 'nodejs20.x');
+      }
+
+      //isComplete Handlers
+      // Handlers
+      const isCompleteHandler = provider.isCompleteHandler as lambda.Function;
+      const isCompleteHandlerRuntime = isCompleteHandler.node.children.find((child) => child instanceof cdk.CfnResource && child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        isCompleteHandlerRuntime.addPropertyOverride('Runtime', 'nodejs20.x');
+      }
+    }
+  }
+
+  private handleReceiptRuleSet(ruleSet: ReceiptRuleSet): void {
+    const functionnode = ruleSet.node.findChild('DropSpam');
+    const ses = functionnode.node.findChild('Function') as lambda.CfnFunction;
+    if (ses.runtime && this.isValidRuntime(ses.runtime)) {
+      ses.addPropertyOverride('Runtime', 'nodejs20.x');
+    }
+  }
+
+  /**
+ *Runtime Validation
+ *
+ */
+  private isValidRuntime(runtime: string) : boolean {
```

Comment:
This is just looking for an equivalent runtime. I think you want this to look for a runtime family. The way it is set up will mean we only replace `nodejs20.x` runtimes with other `nodejs20.x` runtimes. I think you want this to look if the runtime for the visited node is in the runtime family you're trying to replace.